CMSgov/mint-app
# [EASI-3462] Consolidate ExternalLinkModal

# EASI-3462


## Changes and Description

- renamed component to `ExternalLinkWithModal` to be more aptly described
- moved external link modal state logic into its own component



## How to test this change 

1. Add a new document 
2. try the view button
3. verify behavior is as expected



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
i
